q2_phylogeny/plugin_setup.py

```diff
@@ -344,3 +344,57 @@
     description=("Remove features from a feature table if their identifiers "
                  "are not tip identifiers in tree.")
 )
+
+plugin.pipelines.register_function(
+    function=q2_phylogeny.fasttree_with_mafft,
+    inputs={
+        'sequences': FeatureData[Sequence],
+    },
+    parameters={
+        'n_threads': Int % Range(1, None),
+        'max_gap_frequency': Float % Range(0, 1, inclusive_end=True),
+        'min_conservation': Float % Range(0, 1, inclusive_end=True)
+    },
+    outputs=[
+        ('alignment', FeatureData[AlignedSequence]),
+        ('masked_alignment', FeatureData[AlignedSequence]),
+        ('tree', Phylogeny[Unrooted]),
+        ('rooted_tree', Phylogeny[Rooted]),
+    ],
+    input_descriptions={
+        'sequences': 'The sequences to be used for creating a '
+                     'fasttree based rooted phylogenetic tree.'
+    },
+    parameter_descriptions={
+        'n_threads': 'The number of threads. (Use -1 to automatically use all '
+                     'available cores)',
+        'max_gap_frequency':  'The maximum relative frequency of gap '
```

Comment:
should parameters like this be prefixed by the methods that uses them? e.g., this parameter would become `mask_max_gap_frequency`. Two reasons:
1. This makes it clearer to the user where this parameter is impacting the pipeline (clarifying in the description could also achieve this)
2. This would prevent parameter name conflicts (obviously not a problem in this pipeline but this could be a useful convention to begin now before the number of pipelines balloons)

I don't have a strong opinion here, just tossing the idea out there (and at the very least the description should specify where these parameters are applied)

Reply:
I like the idea of prefixing the params, where appropriate.

How about:

`max_gap_frequency -> mask_max_gap_frequency`
`min_conservation -> mask_min_conservation`?

The `n_threads` param can remain as-is.